### PR TITLE
Aic dataset

### PR DESCRIPTION
**Pull request by jin-s13**
add aic dataset

### CODE REVIEW

**Comment by jin-s13** [comment]
Ready to be merged.

File: configs/top_down/resnet/README.md [hunk 1]
```diff
@@ -40,6 +40,12 @@ Following the common setting, the models are trained on COCO train dataset, and
 | [pose_resnet_152](/configs/top_down/resnet/coco/res152_coco_384x288.py) | 384x288 | 0.582 | 0.723 | 0.627 | 0.627 | 0.752 | [ckpt](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmpose/top_down/resnet/res152_coco_384x288-3860d4c9_20200709.pth) | [log](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmpose/top_down/resnet/res152_coco_384x288_20200709.log.json) |
 
 
+### Results on AIC val set.
+
+| Arch  | Input Size | AP | AP<sup>50</sup> | AP<sup>75</sup> | AR | AR<sup>50</sup> | ckpt | log |
+| :-------------- | :-----------: | :------: | :------: | :------: | :------: | :------: |:------: |:------: |
+| [pose_resnet_101](/configs/top_down/resnet/aic/res101_aic_256x192.py) | 256x192 | 0.650 | 0.947 | 0.726 | 0.680 | 0.954 | [ckpt](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmpose/top_down/resnet/res101_aic_256x192-79b35445_20200826.pth) | [log](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmpose/top_down/resnet/res101_aic_256x192_20200826.log.json) |
```
Comment (innerlee) [inline configs/top_down/resnet/README.md:47]: use https://openmmlab.oss-accelerate.aliyuncs.com

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        coco_dt = self.coco.loadRes(res_file)
+        coco_eval = COCOeval(
+            self.coco, coco_dt, 'keypoints', self.sigmas, use_area=False)
+        coco_eval.params.useSegm = None
+        coco_eval.evaluate()
+        coco_eval.accumulate()
+        coco_eval.summarize()
+
+        stats_names = [
+            'AP', 'Ap .5', 'AP .75', 'AP (M)', 'AP (L)', 'AR', 'AR .5',
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:284]: `AP .5`

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+import numpy as np
+from xtcocotools.coco import COCO
+from xtcocotools.cocoeval import COCOeval
+
+from ....core.post_processing import oks_nms, soft_oks_nms
+from ...registry import DATASETS
+from .topdown_coco_dataset import TopDownCocoDataset
+
+
+def _get_mapping_id_name(imgs):
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:13]: can we reuse this function? https://github.com/open-mmlab/mmpose/blob/master/mmpose/datasets/datasets/top_down/topdown_ochuman_dataset.py

Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:13]: or as commented earlier, 
```
id2name = {id:img['file_name'] for id, img in imgs.items()}
name2id= {img['file_name']:id for id, img in imgs.items()}
```

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+        Note:
+            bbox:[x1, y1, w, h]
+        Args:
+            index: coco image id
+        Returns:
+            db entry
+        """
+        im_ann = self.coco.loadImgs(index)[0]
+        width = im_ann['width']
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:141]: when to use `im` and when to use `img`?

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+                    t_s = n_p['keypoints'][n_jt][2]
+                    if t_s > vis_thr:
+                        kpt_score = kpt_score + t_s
+                        valid_num = valid_num + 1
+                if valid_num != 0:
+                    kpt_score = kpt_score / valid_num
+                # rescoring
+                n_p['score'] = kpt_score * box_score
+
+            if self.soft_nms:
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:250]:
```python
nms = soft_oks_nms if self.soft_nms else oks_nms
keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
```

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py (hunk 3 above)
Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:13]: Yes. This func can be reused.

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py (hunk 4 above)
Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_aic_dataset.py]: changed to img_ann.

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        print(f'=> num_images: {self.num_images}')
+        print(f'=> load {len(self.db)} samples')
+
+    def _get_db(self):
+        """Load dataset."""
+        assert self.use_gt_bbox
+        gt_db = self._load_coco_keypoint_annotations()
+        return gt_db
+
+    def _do_python_keypoint_eval(self, res_file):
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:109]: did we changed this name before?

Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:109]: ok we didn't

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        """Keypoint evaluation using COCOAPI."""
+        coco_dt = self.coco.loadRes(res_file)
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:111]: what does `dt` mean, delta time?

File: mmpose/datasets/datasets/top_down/topdown_aic_dataset.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        coco_eval.evaluate()
+        coco_eval.accumulate()
+        coco_eval.summarize()
+
+        stats_names = [
+            'AP', 'AP .5', 'AP .75', 'AP (M)', 'AP (L)', 'AR', 'AR .5',
+            'AR .75', 'AR (M)', 'AR (L)'
+        ]
+
+        info_str = []
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_aic_dataset.py:124]: maybe this pattern can be rewritten (if order of coco_eval.stats and stats_names are same)

```python
info_str = [(name, stat) for name, stat in zip(stats_names, coco_eval.stats)]
```

**Comment by innerlee** [comment]
looks good

**Review by innerlee — APPROVED** [review]

**Pull request merged**